packages/transactions/lib/transactions.js

```diff
@@ -0,0 +1,208 @@
+import {ApiRpc} from '../client/api-rpc';
+import uuid from 'uuid';
+import {getRealm} from '../core/realm';
+import {DOCK_TOKEN_UNIT} from '../core/format-utils';
+import {fetchTransactions} from '../core/subscan';
+import BigNumber from 'bignumber.js';
+import {Accounts} from './accounts';
+import { NetworkManager } from './network-manager';
+
+export const TransactionStatus = {
+  InProgress: 'pending',
+  Failed: 'failed',
+  Complete: 'complete',
+};
+
+/**
+ * Transactions Module
+ */
+export class Transactions {
+  constructor() {
+    this.transactions = [];
+  }
+
+  /**
+   * Get transactions module instance
+   * @returns module instance
+   */
+  static getInstance() {
+    if (!Transactions.instance) {
+      Transactions.instance = new Transactions();
+    }
+
+    return Transactions.instance;
+  }
+  
+  /**
+   * Load external transactions for the given account
+   * 
+   * @param {string} account 
+   */
+  async loadExternalTransactions(account) {
+    const realm = getRealm();
+    const dbTransactions = realm.objects('Transaction').toJSON();
+  
+    const handleTransaction = tx => {
+      if (tx.from !== account && tx.to !== account) {
+        return;
+      }
+  
+      if (dbTransactions.find(item => item.hash === tx.hash)) {
+        return;
+      }
+  
+      const newTx = {
+        amount: BigNumber(tx.amount).times(DOCK_TOKEN_UNIT).toString(),
+        feeAmount: tx.fee,
+        recipientAddress: tx.to,
+        fromAddress: tx.from,
+        id: tx.hash,
+        hash: tx.hash,
+        network: 'mainnet',
```

Comment:
can't network be set to something else? Do we need to match?

Reply:
This code will just be triggered in mainnet, but would make sense to just use the current network id, in case we support additional subscan network in the future.

Reply:
fixed